Assignment5/Graph.java

```diff
@@ -0,0 +1,98 @@
+package Assignment5;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.ArrayList;
+
+public class Graph {
+	int numberEdges;
+	HashSet<Character> nodes;
+	HashMap<Character, HashSet<Character>> adjacentNodes;
+	HashMap<Character, Integer> innerEdges;

```

Comment:
I'd call this "incomingEdges" rather than "inner edges", it makes it clearer what it is doing.